mvlearn/ajive/ajive.py

```diff
@@ -0,0 +1,776 @@
+# Copyright 2019 NeuroData (http://neurodata.io)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import warnings
+from scipy.sparse import issparse
+from copy import deepcopy
+from sklearn.externals.joblib import load, dump
+import pandas as pd
+from mvlearn.utils.utils import check_Xs
+from .block_visualization import _data_block_heatmaps, \
+_ajive_full_estimate_heatmaps
+
+from .utils import svd_wrapper, centering
+from .wedin_bound import get_wedin_samples
+from .random_direction import sample_randdir
+from .pca import pca
+
+
+class ajive(object):
+    """
+    An implementation of Angle-based Joint and Individual Variation Explained.
+    This algorithm takes multiple input views with the same number of samples
+    and decomposes them into 3 distinct matrices representing:
+        - Individual variation of each particular view
+        - Joint variation shared by all views
+        - Noise
+
+    Parameters
+    ----------
+    init_signal_ranks: list or dict
+        The initial signal ranks.
+
+    joint_rank: int = None
+        Rank of the joint variation matrix. If None, will estimate the 
+        joint rank. Otherwise, will use provided joint rank.
+
+    indiv_ranks: list or dict
+        Ranks of individual variation matrices. If None, will estimate the 
+        individual ranks. Otherwise, will use provided individual ranks.
+
+    center: bool = True
+        Boolean for centering matrices.
+
+    reconsider_joint_components: bool = True
+        Triggers _reconsider_joint_components function
+
+    wedin_percentile: int = 5
+        Percentile used for wedin (lower) bound cutoff for squared 
+        singular values used to estimate joint rank.
+
+    n_wedin_samples: int = 1000
+        Number of wedin bound samples to draw.
+
+    precomp_wedin_samples: Dict of array-like or list of array-like
+        Wedin samples that are precomputed for each view.
+
+    randdir_percentile: int = 95
+        Percentile for random direction (lower) bound cutoff for squared
+        singular values used to estimate joint rank.
+
+    n_randdir_samples: int = 1000
+        Number of random direction samples to draw.
+
+    precomp_randdir_samples: array-like = None
+        Precomputed random direction samples.
+
+    n_jobs: int = None
```

Comment:
To give a default, these should all say "int, default=None", not "int = None". Sorry, but you should change this for all your methods if you did it this way